src/PersistentPageIdentifiersExtension.php

```diff
@@ -29,11 +30,21 @@ public function getIdGenerator(): IdGenerator {
 	}
 
 	public function getPersistentPageIdentifiersRepo(): PersistentPageIdentifiersRepo {
+		if ( defined( 'MW_PARSER_TEST' ) ) {
+			return $this->getParserTestPersistentPageIdentifiersRepo();
+		}
```

Comment:
What about doing something similar to https://github.com/ProfessionalWiki/SPARQL/blob/master/src/HookHandlers.php#L25-L44? We could add `setPersistentPageIdentifiersRepo` and call it from that hook.

Reply:
That works.